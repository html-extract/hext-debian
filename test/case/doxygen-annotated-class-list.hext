<tr id^="row_0_">
  <td class="entry">
    <a @text:class_name/>
  </td>
  <td class="desc" @text:class_desc/>
</tr>
