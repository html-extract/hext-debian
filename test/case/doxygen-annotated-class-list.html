<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<meta http-equiv="Content-Type" content="text/xhtml;charset=UTF-8"/>
<meta http-equiv="X-UA-Compatible" content="IE=9"/>
<meta name="viewport" content="width=device-width, initial-scale=1">
<title>libhext: Class List</title>
<script type="text/javascript" src="jquery.js"></script>
<script type="text/javascript" src="dynsections.js"></script>
<link href="search/search.css" rel="stylesheet" type="text/css"/>
<script type="text/javascript" src="search/search.js"></script>
<script type="text/javascript">
  $(document).ready(function() { searchBox.OnSelectItem(0); });
</script>
<link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.4/css/bootstrap.min.css">
<link rel="stylesheet" href="style.css">
</head>
<body>
<div id="top">
<div id="titlearea"></div>
<!-- Generated by Doxygen 1.8.8 -->
<script type="text/javascript">
var searchBox = new SearchBox("searchBox", "search",false,'Search');
</script>
  <div id="navrow1" class="tabs">
    <ul class="tablist">
      <li><a href="index.html"><span>Main&#160;Page</span></a></li>
      <li><a href="namespaces.html"><span>Namespaces</span></a></li>
      <li class="current"><a href="annotated.html"><span>Classes</span></a></li>
      <li><a href="files.html"><span>Files</span></a></li>
      <li>
        <div id="MSearchBox" class="MSearchBoxInactive">
        <span class="left">
          <img id="MSearchSelect" src="search/mag_sel.png"
               onmouseover="return searchBox.OnSearchSelectShow()"
               onmouseout="return searchBox.OnSearchSelectHide()"
               alt=""/>
          <input type="text" id="MSearchField" value="Search" accesskey="S"
               onfocus="searchBox.OnSearchFieldFocus(true)" 
               onblur="searchBox.OnSearchFieldFocus(false)" 
               onkeyup="searchBox.OnSearchFieldChange(event)"/>
          </span><span class="right">
            <a id="MSearchClose" href="javascript:searchBox.CloseResultsWindow()"><img id="MSearchCloseImg" border="0" src="search/close.png" alt=""/></a>
          </span>
        </div>
      </li>
    </ul>
  </div>
  <div id="navrow2" class="tabs2">
    <ul class="tablist">
      <li class="current"><a href="annotated.html"><span>Class&#160;List</span></a></li>
      <li><a href="inherits.html"><span>Class&#160;Hierarchy</span></a></li>
      <li><a href="functions.html"><span>Class&#160;Members</span></a></li>
    </ul>
  </div>
</div><!-- top -->
<!-- window showing the filter options -->
<div id="MSearchSelectWindow"
     onmouseover="return searchBox.OnSearchSelectShow()"
     onmouseout="return searchBox.OnSearchSelectHide()"
     onkeydown="return searchBox.OnSearchSelectKey(event)">
<a class="SelectItem" href="javascript:void(0)" onclick="searchBox.OnSelectItem(0)"><span class="SelectionMark">&#160;</span>All</a><a class="SelectItem" href="javascript:void(0)" onclick="searchBox.OnSelectItem(1)"><span class="SelectionMark">&#160;</span>Classes</a><a class="SelectItem" href="javascript:void(0)" onclick="searchBox.OnSelectItem(2)"><span class="SelectionMark">&#160;</span>Namespaces</a><a class="SelectItem" href="javascript:void(0)" onclick="searchBox.OnSelectItem(3)"><span class="SelectionMark">&#160;</span>Files</a><a class="SelectItem" href="javascript:void(0)" onclick="searchBox.OnSelectItem(4)"><span class="SelectionMark">&#160;</span>Functions</a><a class="SelectItem" href="javascript:void(0)" onclick="searchBox.OnSelectItem(5)"><span class="SelectionMark">&#160;</span>Variables</a><a class="SelectItem" href="javascript:void(0)" onclick="searchBox.OnSelectItem(6)"><span class="SelectionMark">&#160;</span>Typedefs</a><a class="SelectItem" href="javascript:void(0)" onclick="searchBox.OnSelectItem(7)"><span class="SelectionMark">&#160;</span>Enumerations</a><a class="SelectItem" href="javascript:void(0)" onclick="searchBox.OnSelectItem(8)"><span class="SelectionMark">&#160;</span>Enumerator</a><a class="SelectItem" href="javascript:void(0)" onclick="searchBox.OnSelectItem(9)"><span class="SelectionMark">&#160;</span>Pages</a></div>

<!-- iframe showing the search results (closed by default) -->
<div id="MSearchResultsWindow">
<iframe src="javascript:void(0)" frameborder="0" 
        name="MSearchResults" id="MSearchResults">
</iframe>
</div>

<div class="header">
  <div class="headertitle">
<div class="title">Class List</div>  </div>
</div><!--header-->
<div class="contents">
<div class="textblock">Here are the classes, structs, unions and interfaces with brief descriptions:</div><div class="directory">
<div class="levels">[detail level <span onclick="javascript:toggleLevel(1);">1</span><span onclick="javascript:toggleLevel(2);">2</span>]</div><table class="directory">
<tr id="row_0_" class="even"><td class="entry"><span style="width:0px;display:inline-block;">&#160;</span><span id="arr_0_" class="arrow" onclick="toggleFolder('0_')">&#9660;</span><span class="icona"><span class="icon">N</span></span><a class="el" href="namespacehext.html" target="_self">hext</a></td><td class="desc">The hext namespace is the top level namespace for everything in libhext </td></tr>
<tr id="row_0_0_"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="classhext_1_1AttributeCapture.html" target="_self">AttributeCapture</a></td><td class="desc">Captures an HTML Element's attribute </td></tr>
<tr id="row_0_1_" class="even"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="classhext_1_1AttributeCountMatch.html" target="_self">AttributeCountMatch</a></td><td class="desc">Matches HTML elements that have a certain amount of HTML attributes </td></tr>
<tr id="row_0_2_"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="classhext_1_1AttributeMatch.html" target="_self">AttributeMatch</a></td><td class="desc">Matches HTML elements having an HTML attribute with a certain name and, optionally, whose value is matched by a <a class="el" href="classhext_1_1ValueTest.html" title="Abstract base for every ValueTest. ">ValueTest</a> </td></tr>
<tr id="row_0_3_" class="even"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="classhext_1_1BeginsWithTest.html" target="_self">BeginsWithTest</a></td><td class="desc">Tests whether a string begins with a given literal </td></tr>
<tr id="row_0_4_"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="classhext_1_1Capture.html" target="_self">Capture</a></td><td class="desc">Abstract base for every <a class="el" href="classhext_1_1Capture.html" title="Abstract base for every Capture. ">Capture</a> </td></tr>
<tr id="row_0_5_" class="even"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="classhext_1_1ChildCountMatch.html" target="_self">ChildCountMatch</a></td><td class="desc">Matches HTML elements that have a certain amount of children of type element (exluding text nodes, document nodes and others) </td></tr>
<tr id="row_0_6_"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="classhext_1_1ContainsTest.html" target="_self">ContainsTest</a></td><td class="desc">Tests whether a string contains a given literal </td></tr>
<tr id="row_0_7_" class="even"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="classhext_1_1ContainsWordsTest.html" target="_self">ContainsWordsTest</a></td><td class="desc">Tests whether a string contains all given words </td></tr>
<tr id="row_0_8_"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="classhext_1_1EndsWithTest.html" target="_self">EndsWithTest</a></td><td class="desc">Tests whether a string ends with a given literal </td></tr>
<tr id="row_0_9_" class="even"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="classhext_1_1EqualsTest.html" target="_self">EqualsTest</a></td><td class="desc">Tests whether a string equals a given literal </td></tr>
<tr id="row_0_10_"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="classhext_1_1Extractor.html" target="_self">Extractor</a></td><td class="desc">Extracts ResultTrees from HTML </td></tr>
<tr id="row_0_11_" class="even"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="classhext_1_1FunctionCapture.html" target="_self">FunctionCapture</a></td><td class="desc">Captures the result of applying a function to an HTML node </td></tr>
<tr id="row_0_12_"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="classhext_1_1FunctionMatch.html" target="_self">FunctionMatch</a></td><td class="desc">Matches if the result of applying a given CaptureFunction to an HTML node passes a <a class="el" href="classhext_1_1ValueTest.html" title="Abstract base for every ValueTest. ">ValueTest</a> </td></tr>
<tr id="row_0_13_" class="even"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="classhext_1_1Html.html" target="_self">Html</a></td><td class="desc">A RAII wrapper for Gumbo </td></tr>
<tr id="row_0_14_"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="classhext_1_1Match.html" target="_self">Match</a></td><td class="desc">Abstract base for every <a class="el" href="classhext_1_1Match.html" title="Abstract base for every Match. ">Match</a> </td></tr>
<tr id="row_0_15_" class="even"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="classhext_1_1NegateMatch.html" target="_self">NegateMatch</a></td><td class="desc">Matches HTML nodes for which every given <a class="el" href="classhext_1_1Match.html" title="Abstract base for every Match. ">Match</a> returns false </td></tr>
<tr id="row_0_16_"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="classhext_1_1NegateTest.html" target="_self">NegateTest</a></td><td class="desc">Negates the result of another <a class="el" href="classhext_1_1ValueTest.html" title="Abstract base for every ValueTest. ">ValueTest</a> </td></tr>
<tr id="row_0_17_" class="even"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="classhext_1_1NthChildMatch.html" target="_self">NthChildMatch</a></td><td class="desc">Matches HTML nodes having a certain position within their parent HTML element </td></tr>
<tr id="row_0_18_"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="classhext_1_1OnlyChildMatch.html" target="_self">OnlyChildMatch</a></td><td class="desc">Matches HTML nodes that are the only child of their parent HTML element </td></tr>
<tr id="row_0_19_" class="even"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="classhext_1_1RegexTest.html" target="_self">RegexTest</a></td><td class="desc">Tests whether another string matches a given regex </td></tr>
<tr id="row_0_20_"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="classhext_1_1ResultTree.html" target="_self">ResultTree</a></td><td class="desc">A self-managing tree containing all values captured from HTML </td></tr>
<tr id="row_0_21_" class="even"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="classhext_1_1Rule.html" target="_self">Rule</a></td><td class="desc">Extracts values from HTML </td></tr>
<tr id="row_0_22_"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="classhext_1_1SyntaxError.html" target="_self">SyntaxError</a></td><td class="desc">The exception that is thrown when parsing invalid hext </td></tr>
<tr id="row_0_23_" class="even"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="classhext_1_1TextNodeMatch.html" target="_self">TextNodeMatch</a></td><td class="desc">Matches if an HTML node is a text node </td></tr>
<tr id="row_0_24_"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="classhext_1_1ValueTest.html" target="_self">ValueTest</a></td><td class="desc">Abstract base for every <a class="el" href="classhext_1_1ValueTest.html" title="Abstract base for every ValueTest. ">ValueTest</a> </td></tr>
</table>
</div><!-- directory -->
</div><!-- contents -->
<div id="nav-path" class="navpath"><!-- id is needed for treeview function! -->
  <ul>
    <li class="footer">Generated on Tue Jul 21 2015 22:16:33 for libhext by
    <a href="http://www.doxygen.org/index.html">doxygen</a> 1.8.8 </li>
  </ul>
</div>
</body>
</html>
