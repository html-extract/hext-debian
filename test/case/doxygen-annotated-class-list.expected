{"class_desc":"Abstract base for every Capture","class_name":"Capture"}
{"class_desc":"Abstract base for every Match","class_name":"Match"}
{"class_desc":"Abstract base for every ValueTest","class_name":"ValueTest"}
{"class_desc":"A RAII wrapper for Gumbo","class_name":"Html"}
{"class_desc":"A self-managing tree containing all values captured from HTML","class_name":"ResultTree"}
{"class_desc":"Captures an HTML Element's attribute","class_name":"AttributeCapture"}
{"class_desc":"Captures the result of applying a function to an HTML node","class_name":"FunctionCapture"}
{"class_desc":"Extracts ResultTrees from HTML","class_name":"Extractor"}
{"class_desc":"Extracts values from HTML","class_name":"Rule"}
{"class_desc":"Matches HTML elements having an HTML attribute with a certain name and, optionally, whose value is matched by a ValueTest","class_name":"AttributeMatch"}
{"class_desc":"Matches HTML elements that have a certain amount of children of type element (exluding text nodes, document nodes and others)","class_name":"ChildCountMatch"}
{"class_desc":"Matches HTML elements that have a certain amount of HTML attributes","class_name":"AttributeCountMatch"}
{"class_desc":"Matches HTML nodes for which every given Match returns false","class_name":"NegateMatch"}
{"class_desc":"Matches HTML nodes having a certain position within their parent HTML element","class_name":"NthChildMatch"}
{"class_desc":"Matches HTML nodes that are the only child of their parent HTML element","class_name":"OnlyChildMatch"}
{"class_desc":"Matches if an HTML node is a text node","class_name":"TextNodeMatch"}
{"class_desc":"Matches if the result of applying a given CaptureFunction to an HTML node passes a ValueTest","class_name":"FunctionMatch"}
{"class_desc":"Negates the result of another ValueTest","class_name":"NegateTest"}
{"class_desc":"Tests whether another string matches a given regex","class_name":"RegexTest"}
{"class_desc":"Tests whether a string begins with a given literal","class_name":"BeginsWithTest"}
{"class_desc":"Tests whether a string contains a given literal","class_name":"ContainsTest"}
{"class_desc":"Tests whether a string contains all given words","class_name":"ContainsWordsTest"}
{"class_desc":"Tests whether a string ends with a given literal","class_name":"EndsWithTest"}
{"class_desc":"Tests whether a string equals a given literal","class_name":"EqualsTest"}
{"class_desc":"The exception that is thrown when parsing invalid hext","class_name":"SyntaxError"}
{"class_desc":"The hext namespace is the top level namespace for everything in libhext","class_name":"hext"}
